src/det.jl

```diff
@@ -21,6 +21,23 @@ end
     return vecdot(x0, cross(x1, x2))
 end
 
+@inline function _det(::Size{(4,4)}, A::StaticMatrix,S::Type)
+    A = similar_type(A,S)(A)
```

Comment:
By now, I am used to the "SSA" form so I do tend to write `a2 = f(a); a3 = g(a2)` etc. In this case something like `A_S` would look clear. 

Dispatch might be a very clean solution - the line won't even appear when `S` is already correct, which may be important for e.g. heap allocated static arrays or if `similar_type` always changes types or if the compiler doesn't optimiza away the construction for some reason. 

So let's try dispatch!